src/NuGet.Core/NuGet.LibraryModel/LibraryIncludeFlagUtils.cs

```diff
@@ -118,5 +118,23 @@ public static LibraryIncludeFlags GetFlags(string flags, LibraryIncludeFlags def
 
             return result;
         }
+
+        private static readonly Dictionary<LibraryIncludeFlags, string> LibraryIncludeFlagsCache = new();
+
+        /// <summary>
+        /// Efficiently converts <see cref="LibraryIncludeFlags"/> to it's <see cref="string"/> representation.
+        /// </summary>
+        /// <param name="includeFlags">The <see cref="LibraryIncludeFlags"/> instance to get the <see cref="string"/> representation for.</param>
+        /// <returns>The <see cref="string"/> representation of <paramref name="includeFlags"/>.</returns>
+        public static string AsString(this LibraryIncludeFlags includeFlags)
+        {
+            if (!LibraryIncludeFlagsCache.TryGetValue(includeFlags, out string enumAsString))
+            {
+                enumAsString = includeFlags.ToString();
+                LibraryIncludeFlagsCache[includeFlags] = enumAsString;
```

Comment:
This is not threadsafe, and Nuget restores projects in parallel. This will cause problems.